### DISCUSSION TITLE
Interpreting ContactNormalB value returned by getContactPoints

### DISCUSSION THREAD

**Discussion by shivanimall** [bulletphysics/bullet3#3800] [General]
I tried multiple objects to simulate collision, and also created angular and linear velocities along all axes using resetBaseVelocity. However, when I try to get the contact collision forces, the third value(Z direction) is only populated(with -1) in the vector returned by `ContactNormalB` as following: (-0, -0, -1) from `getContactPoints`. For the first 2 values, I am always getting 0.

I could see frictional forces along X/Y directions. But, how should I interpret the value of  `ContactNormalB`? From my understanding if an object falls at an angle, there should be some force values along all axes(X, Y, Z). So are there no contact normal forces along X/Y directions? which seems incorrect.

Please let me know if my question is unclear. Thanks

**Comment by shivanimall** [comment]
Hi @erwincoumans following up on this, in case my question is unclear, please let me know.

**Reply by xissburg** [reply]
The contact normal B should be a vector in B's object space. To obtain it in world space you have to apply B's rotation to it. I know this on a conceptual level, not sure what the code would look like in your situation but I think you can figure it out with this information.

**Reply by erwincoumans** [reply]
ContactNormalB is in world space, pointing from B towards A. If an object falls at an angle at a flat surface, the contact normal can still be [0,0,-1], since the contact is generated from a vertex-face contact. There are no normal forces along X/Y, only friction forces along X/Y.

**Comment by shivanimall** [comment]
thanks for the response @erwincoumans 
noted the point on vertex-face contact. is it possible to point me to a demo/example that can generate normal forces along X/Y directions as well? any simple free fall object-collision simulation on the plane would be great.

hoping to see simulation cases that I am not capturing in my examples so that I calibrate my use case. thanks again.
